### PR TITLE
Fix failed "closestPoint" for space point to bspline when the closest point is at a bezier junction

### PR DESCRIPTION
**Pull request by EarlinLutz**
* Space point projects to a bspline exactly at a knot where there is a transition between 2 beziers.
* (A-X) dot X' = 0 should have a root on both beziers.
* but numerically the incoming bezier coffs are (-23, -18,  ..., -1.3e-9)  (+1.5e-9, 5, 10, .....)
* Add simple proximity test at the endpoint.

### CODE REVIEW

**Review by dassaf4 — APPROVED** [review]

File: core/geometry/src/test/curve/CurveChainWithDistanceIndex.test.ts [hunk 1]
```diff
@@ -43,4 +44,60 @@ describe("CurveChainWithDistanceIndex", () => {
     GeometryCoreTestIO.saveGeometry(allGeometry, "CurveChainWithDistanceIndex", "ClosestPointProblem");
     expect(ck.getNumErrors()).equals(0);
   });
-});
+  const pathWithBSplineProblemFile = "./src/test/testInputs/CurveChainWithDistanceIndex/WoodFieldPath/pathWithBSplines.imjs";
+
+    it.only("PathWithBsplineLength", () => {
```
[the remark this answers is not in the document]
Comment (EarlinLutz) [reply core/geometry/src/test/curve/CurveChainWithDistanceIndex.test.ts]: (thank you Paul & David ...)

File: core/geometry/src/test/curve/CurveChainWithDistanceIndex.test.ts [hunk 2]
```diff
@@ -43,4 +43,60 @@ describe("CurveChainWithDistanceIndex", () => {
...
+  const pathWithBSplineProblemFile = "./src/test/testInputs/CurveChainWithDistanceIndex/WoodfieldPath/pathWithBsplines.imjs";
+
+    it("PathWithBsplineLength", () => {
+      const ck = new Checker();
+      const allGeometry: GeometryQuery[] = [];
+
+      const stringData = fs.readFileSync(pathWithBSplineProblemFile, "utf8");
+      if (stringData) {
+        const pathObject = IModelJson.Reader.parse(JSON.parse(stringData));
+        if (pathObject) {
```
Comment (a-gagnon) [inline core/geometry/src/test/curve/CurveChainWithDistanceIndex.test.ts:55]: Wouldn't it be better to `expect` or `assert` on those? It if ever happens, it'll simply skip the test silently.

Comment (dassaf4) [reply core/geometry/src/test/curve/CurveChainWithDistanceIndex.test.ts]: Yes indeed! I've fixed. Thanks!

**Comment by pmconne** [comment]
Force-merging because unrelated test failures.

**Pull request merged**